python/lsst/ip/isr/isrTask.py

```diff
@@ -943,7 +969,7 @@ def readIsrData(self, dataRef, rawExposure):
                 # If using a new-style kernel, always use the self-consistent
                 # gains, i.e. the ones inside the kernel object itself
                 brighterFatterKernel = dataRef.get("brighterFatterKernel")
-                brighterFatterGains = brighterFatterKernel.gain
+                brighterFatterGains = getattr(brighterFatterKernel, 'gain', None)
```

Comment:
not that its bad, but why getattr here? if it is a new style then gain will be an attribute, if its an old style an exception will be thrown (likely by the dataRef.get) and this block wont be executed. I would be more worried about getting this far if it is an old style. If you wanted to be safe, then maybe a different except block? This seems like it will just hide a problem

Reply:
There is a possibility that there will be no .gains attribute, depending on how the kernel was generated.  I will ask @mfisherlevine if this is still possible.

Reply:
In short, yes, Chris is right, that _is_ technically possible. However, I think that's a bug. The generating code always has them, and downstream code should be able to rely upon them being stored in the kernel so the gains can be applied in a consistent manner, so I think the best course of action here is to allow yourselves to rely up this, and ticket fixing up the BF code so it always stores the gain. The patch should be trivial. Does that sound OK?

Reply:
DM-23273 requests that gain existence be enforced.  Marking this done, and will merge after testing.